lib/mosaic-routing/src/main/java/org/eclipse/mosaic/lib/routing/CandidateRoute.java

```diff
@@ -23,25 +23,94 @@
  */
 public class CandidateRoute {
 
+    /**
+     * The list of connections forming this route.
+     */
     private final List<String> connectionIds;
+
+    /**
+     * The length in meters of this route.
+     */
     private final double length;
+
+    /**
+     * The approximated driving time in seconds on this route.
+     */
     private final double time;
 
+    /**
+     * The distance in meters from the start node of the first connection in the connectionIds list, to
+     * the point the source query was issued.
+     */
+    private final double offsetFromSource;
+
+    /**
+     * The distance in meters from the point the target query was issued, until the end node of the
+     * final connection in the connectionIds list.
+     */
+    private final double offsetToTarget;
+
+    /**
+     * @param connectionIds the list of connection id this route persists of.
+     * @param length the length in m of this route.
+     * @param time the approximate travel time in seconds of this route.
+     */
     public CandidateRoute(List<String> connectionIds, double length, double time) {
+        this(connectionIds, length, time, 0, 0);
+    }
+
+    /**
+     *
+     * @param connectionIds the list of connection id this route persists of.
+     * @param length the length in m of this route.
+     * @param time the approximate travel time in seconds of this route.
+     * @param offsetFromSource the distance in meters from the start node of the first connection in the connectionIds list, to
+     *                         the point the source query was issued.
+     * @param offsetToTarget the distance in meters from the point the target query was issued, until the end node of the
+     *                       final connection in the connectionIds list.
+     */
+    public CandidateRoute(List<String> connectionIds, double length, double time, double offsetFromSource, double offsetToTarget) {
         this.connectionIds = connectionIds;
         this.length = length;
         this.time = time;
+        this.offsetFromSource = offsetFromSource;
+        this.offsetToTarget = offsetToTarget;
     }
 
+    /**
+     * @return the list of connection ids forming this route.
```

Comment:
I think these type of javadocs leads to spotbugs warnings, as the actual documentation is missing.
Personally, I think the documentation for getters should either be really verbose (like the member variable) or none.